packages/components/src/components/Button/types.ts

```diff
@@ -0,0 +1,12 @@
+import { ButtonProps } from "./Button";
+
+export type ToggleButtonContextType =
+    | {
+          /**
+           * Boolean value indicating whether or not the contexed button is selected or not.
+           */
+          isSelected: boolean;
+      }
+    | undefined;
+
+export type ButtonGroupChildrenType = React.ReactElement<ButtonProps> | boolean | null | undefined;
```

Comment:
It seems like this type is similar to `TabsChildrenType` and there will probably be new identical types. I would like a generic type `ChildrenType` where you enter `ButtonProps` into. Also having `| boolean | null | undefined` can be replaced by `ReactNode`. I would also move this type to where it belongs `ButtonGroup.tsx`.